client/src/components/DisplayAllCampaigns.tsx

```diff
@@ -0,0 +1,34 @@
+import { Button, Flex } from '@radix-ui/themes';
+import { useEffect, useState } from 'react';
+import axios from 'axios';
+import { useNavigate } from 'react-router-dom';
+import { settlementApi } from '../service/api';
+import { TypeServerSettlement } from '../../../SettlementTypes';
+
+const DisplayAllCampaigns = () => {
+    const navigate = useNavigate();
+    const [allCampaigns, setAllCampaigns] = useState<TypeServerSettlement[]>([]);
+
+    useEffect(() => {
```

Comment:
Also, while [axios-hooks](https://www.npmjs.com/package/axios-hooks) isn't the greatest approach, it's better than what you're doing, as there's a lot of cases you need to handle here.

Reply:
got it working in both spots where axios calls are being made